src/transfer-gateway.ts

```diff
@@ -0,0 +1,71 @@
+import BN from 'bn.js'
+import { Client } from './client'
+import { Contract } from './contract'
+import { Address } from './address'
+import {
+  TransferGatewayWithdrawERC721Request,
+  TransferGatewayWithdrawalReceiptRequest,
+  TransferGatewayWithdrawalReceiptResponse,
+  TransferGatewayWithdrawalReceipt
+} from './proto/loom_pb'
+import { marshalBigUIntPB, unmarshalBigUIntPB } from './big-uint'
+
+export interface ITransferGatewayReceipt {
+  tokenOwner: Address
+  // Mainnet address of token contract
+  tokenContract: Address
+  transferGatewayKind: Number
+  // ERC721 token ID, or amount of ERC20/ETH
+  value: BN
+  withdrawalNonce: Number
+  // Signature generated by the Oracle that confirmed the withdrawal
+  oracleSignature: Uint8Array
+}
+
+export class TransferGateway {
+  private _transferGatewayContract!: Contract
+
+  static async createAsync(client: Client, callerAddr: Address): Promise<TransferGateway> {
+    const contractAddr = await client.getContractAddressAsync('transfergateway')
+    if (!contractAddr) {
+      throw Error('Failed to resolve contract address for TransferGateway')
+    }
+
+    const contract = new Contract({ contractAddr, callerAddr, client })
+    return new TransferGateway(contract)
+  }
+
+  constructor(contract: Contract) {
+    this._transferGatewayContract = contract
+  }
+
+  async withdrawERC721Async(tokenId: BN, tokenContract: Address): Promise<void> {
+    const tgWithdrawERC721Req = new TransferGatewayWithdrawERC721Request()
```

Comment:
lol, just call it `req`, why the long name, there's only one request in this function

Reply:
I like long names, important people has long names :D

Reply:
Because of there more requests on the entire class (also important people has long names lol)